### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by georgemorgan**
**Almost ready to merge.**

Miscellaneous cleanups here and there. Mostly just reorganizing headers.

Compilation is a bit faster now. Warnings have been eliminated.

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -29,20 +29,23 @@ ARM_SRC_DIRS := carbon/atsam4s										\
 				runtime/arch/armv7									\
 				runtime/src
 
-ARM_CFLAGS	 := -std=gnu99											\
+ARM_CFLAGS	 := -std=c99											\
 				-Wall												\
 				-Wextra												\
```
Comment (georgemorgan) [inline makefile]: I will try to fix this.

File: assets/flipper.mk [hunk 2]
```diff
@@ -19,28 +19,26 @@ FLIPPER_LD := arm-none-eabi-gcc
...
-	-DPLATFORM_HEADER="<flipper/atsam4s/atsam4s.h>"
-
-FLIPPER_LDFLAGS := \
-	-mcpu=cortex-m4 \
-	-g \
-	-ffreestanding \
-	-nostdlib \
-	-fPIC \
-	-Os
+ARM_CFLAGS := -std=c99											\
```
Comment (georgemorgan) [inline assets/flipper.mk]: I will try to fix this formatting.

**Comment by nicholastmosher** [comment]
Just pulled this on my machine and everything builds 👍

**Comment by nicholastmosher** [comment]
@georgemorgan Was this not building when you merged it?

**Comment by georgemorgan** [comment]
Nah, I canceled the build because it was just a makefile change that Travis doesn't test anyway.

**Pull request merged**